doc/developer-notes.md

```diff
@@ -477,34 +477,34 @@ class A
   `int8_t`. Do not use bare `char` unless it is to pass to a third-party API.
   This type can be signed or unsigned depending on the architecture, which can
   lead to interoperability problems or dangerous conditions such as
-  out-of-bounds array accesses
+  out-of-bounds array accesses.
 
-- Prefer explicit constructions over implicit ones that rely on 'magical' C++ behavior
+- Prefer explicit constructions over implicit ones that rely on 'magical' C++ behavior.
 
   - *Rationale*: Easier to understand what is happening, thus easier to spot mistakes, even for those
-  that are not language lawyers
+  that are not language lawyers.
 
 Strings and formatting
 ------------------------
 
 - Be careful of `LogPrint` versus `LogPrintf`. `LogPrint` takes a `category` argument, `LogPrintf` does not.
 
   - *Rationale*: Confusion of these can result in runtime exceptions due to
-    formatting mismatch, and it is easy to get wrong because of subtly similar naming
+    formatting mismatch, and it is easy to get wrong because of subtly similar naming.
 
-- Use `std::string`, avoid C string manipulation functions
+- Use `std::string`, avoid C string manipulation functions.
 
   - *Rationale*: C++ string handling is marginally safer, less scope for
-    buffer overflows and surprises with `\0` characters. Also some C string manipulations
-    tend to act differently depending on platform, or even the user locale
+    buffer overflows and fewer surprises with `\0` characters. Also some C string manipulations
```

Comment:
First sentence still sounds odd to me. How about:

```
C++ string handling is marginally safer, has less scope for buffer overflows and there are fewer surprises with `\0` characters.
```

?

Reply:
Is this okay?

C++ string handling is marginally safer, with less scope for buffer overflows and fewer surprises with `\0` characters.